### ISSUE TITLE
unsupport http version http/1.0

### ISSUE THREAD

**Issue by chppppp** [aaugustin/websockets#816]
```
Traceback (most recent call last):
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/client.py", line 101, in read_http_response
    status_code, reason, headers = await read_response(self.reader)
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/http.py", line 149, in read_response
    raise ValueError(f"unsupported HTTP version: {d(version)}")
ValueError: unsupported HTTP version: HTTP/1.0

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "blindsqli.py", line 9, in <module>
    asyncio.get_event_loop().run_until_complete(
  File "/usr/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "blindsqli.py", line 5, in hello
    async with websockets.connect(uri) as websocket:
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/client.py", line 517, in __aenter__
    return await self
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/client.py", line 542, in __await_impl__
    await protocol.handshake(
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/client.py", line 290, in handshake
    status_code, response_headers = await self.read_http_response()
  File "/home/hook/.local/lib/python3.8/site-packages/websockets/client.py", line 103, in read_http_response
    raise InvalidMessage("did not receive a valid HTTP response") from exc
websockets.exceptions.InvalidMessage: did not receive a valid HTTP response
```

**Comment by aaugustin** [comment]
There's no point supporting HTTP/1.0 in websockets because the WebSocket protocol only works over HTTP/1.1 (and later).

**Comment by twisteroidambassador** [comment]
There actually is a point: health checks.

The documentation describes a health check implemented using `process_request`: https://websockets.readthedocs.io/en/stable/faq/server.html#how-do-i-implement-a-health-check 
This only works for HTTP 1.1 clients. I recently encountered load balancers that send HTTP 1.0 health check requests, and those results in ugly exceptions from websockets, and failed health checks.

**Comment by aaugustin** [comment]
The ugly exceptions will probably be fixed in 17.0, thanks to 81a29b79efc8241a3a287584a56825aa60d1cba4.

HTTP/1.0 health checks will still fail; websockets will reject them before they reach process_request. As a short term mitigation, you can use TCP health checks.

There's an argument for building limited support for HTTP/1.0, similar to what exists for proxies. It's just annoying to add overhead for all users because of legacy hardware :-/
